src/app/components/configuration/configuration.service.js

```diff
@@ -0,0 +1,58 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('gcloudConsole')
+    .service('configuration', Configuration);
+
+  function Configuration($q, GData, firebase) {
+    this.userId = GData.getUser().id;
+    this.$q = $q;
+
+    var url = 'https://phoenix-storage.firebaseio.com/configurations';
+    this.getRef = firebase.getRef(url);
+  }
+
+  Configuration.prototype.getByName = function(name) {
+    var deferred = this.$q.defer();
+
+    this.getRef.then(function(configRef) {
+      configRef.child(name).once('value', function(snap) {
+        var configuration = snap.val();
+
+        if (!configuration) {
+          deferred.reject(new Error('Configuration not found.'));
+          return;
+        }
+
+        deferred.resolve(configuration);
+      });
+    });
+
+    return deferred.promise;
+  };
```

Comment:
Promises look good, sucks we have to wrap an event emitter. My one suggestion would be to return `this.getRef(...)` instead of wrapping it. I believe right now if `getRef` failed, the promise returned from this would never resolve. You *could* fix that by creating a failure callback, but if you simply returned `getRef` and created/returned a new promise within the success block, then you're failure states would all be handled without an additional callback.